File: py/exerc011.py
#**********************************************************************
# FILE NAME   : EXERC011                                              *
# DATE        : 2025-03-27                                            *
# AUTHOR      : FABIO MARQUES (FMARQUES@FMARQUES.ETI.BR)              *
# DATA CENTER : COMPANY.EDUC360                                       *
# PURPOSE     : SUPERMARKET CHECKOUT SIMULATION                       *
#**********************************************************************

# Variable declaration
descricao = ""
valor     = 0
qtd       = 0
total     = 0

while True:
    # Reading n value
    descricao = input("\nInforme a descrição do produto: ")

    if descricao != "":
        # Reading product value
        valor = float(input("Informe o valor do produto: "))
    
        # Reading product quantity
        qtd = float(input("Informe a quantidade: "))
    
        # Calculating total
        total = total + (valor * qtd)
    else: # Running condition ( descricao != 0 )
        break

# Display
print("\nTotal da compra: ", total)

File: js/exerc011.js
/**********************************************************************
* FILE NAME   : EXERC011                                              *
* DATE        : 2025-03-29                                            *
* AUTHOR      : FABIO MARQUES (FMARQUES@FMARQUES.ETI.BR)              *
* DATA CENTER : COMPANY.EDUC360                                       *
* PURPOSE     : SUPERMARKET CHECKOUT SIMULATION                       *
**********************************************************************/

// Variable declaration
let descricao = ""
let valor     = 0
let qtd       = 0
let total     = 0

do {
  // Reading product description
  descricao = prompt("Informe a descrição do produto")
  
  if (descricao != "") {
    // Reading product value
    valor = parseFloat(prompt("Informe o valor do produto"))
    // Reading product quantity
    qtd   = parseInt(prompt("Informe a quantidade"))
    
    // Calculating total
    total += valor * qtd
  }
} while (descricao != "") // Running condition

// Display total
console.log("Total da compra: " + total)